test/puppetlabs/http/client/sync_plaintext_test.clj

```diff
@@ -63,10 +63,12 @@
           (let [request-options (SimpleRequestOptions. (URI. "http://localhost:10000/hello/"))
                 response (java-method request-options)]
             (is (= 200 (.getStatus response)))
+            (is (= "OK" (.getReasonPhrase response)))
             (is (= "Hello, World!" (slurp (.getBody response))))))
         (testing "clojure sync client"
           (let [response (clj-fn "http://localhost:10000/hello/")]
             (is (= 200 (:status response)))
+            (is (= "OK" (:reason-phrase response)))
             (is (= "Hello, World!" (slurp (:body response))))))))))
 
 (deftest sync-client-head-test
```

Comment:
I know this is totally separate from this PR, so you can ignore me, but looks like this whole test can just be refactored to call `basic-test` like the tests below. (and then the 404 could go in here)

Reply:
Well the 404 needs a different url so it can't use `basic-test`. Kinda like keeping this one separate more now that I've added the 404 to this test. It's also a good smoke test: the tests that put assertions in a function like that can be harder to debug, so having one be separate is kinda nice IMO.